documentation/examples/prometheus-uyuni.yml

```diff
@@ -0,0 +1,36 @@
+# A example scrape configuration for running Prometheus with Uyuni.
+
+scrape_configs:
+
+  # Make Prometheus scrape itself for metrics.
+  - job_name: 'prometheus'
+    static_configs:
+      - targets: ['localhost:9090']
+
+  # Discover Uyuni managed targets to scrape.
+  - job_name: 'uyuni'
+
+    # Scrape Uyuni itself to discover new services.
+    uyuni_sd_configs:
+      - host: http://uyuni-project.org
+        username: gopher
+        password: hole
+    relabel_configs:
+    - source_labels: [__meta_uyuni_exporter]
+      target_label: exporter
+    - source_labels: [__meta_uyuni_groups]
+      target_label: groups
+    - source_labels: [__meta_uyuni_minion_hostname]
+      target_label: hostname
+    - source_labels: [__meta_uyuni_primary_fqdn]
+      regex: (.+)
+      target_label: hostname
+    - source_labels: [hostname, __address__]
```

Comment:
Why is this rule needed?

Reply:
We've got two labels specifying the hostname:

- `__meta_uyuni_minion_hostname`: always non-empty
- `__meta_uyuni_primary_fqdn`: can be _null_

We want to point the target to _FQDN:port_ if metadata available, or to _hostname:port_ otherwise.